snuba/query/processors/tagsmap.py

```diff
@@ -88,13 +98,37 @@ def __is_optimizable(
         return None
 
     def process_query(self, query: Query, request_settings: RequestSettings) -> None:
-        enabled = state.get_config("optimize_nested_col_conditions", 0)
-        if not enabled:
-            return
         conditions = query.get_conditions()
         if not conditions:
             return
 
+        # Enable the processor only if we have enough data in the flattened
+        # columns. Which have been deployed at BEGINNING_OF_TIME. If the query
+        # starts earlier than that we do not apply the optimization.
+        from_time_condition = list(
+            filter(
+                lambda c: is_condition(c)
+                and c[0] == self.__start_ts_col
+                and c[1] in (">=", ">")
+                and isinstance(c[2], str),
+                conditions,
+            )
+        )
+        if not from_time_condition:
+            return
+        try:
+            start_ts = datetime.strptime(from_time_condition[0][2], "%Y-%m-%dT%H:%M:%S")
```

Comment:
Shouldn't this be the minimum value of `[parse_datetime(c[2]) for c in from_time_condition]`?

Reply:
True, we should never have two separate start_ts conditions though. Will fix

Reply:
Actually the greatest value is the one to use.

WHERE time > 1970-01-01 AND time > 1971-01-01 
is equivalent to 
WHERE  time > 1971-01-01

Reply:
Oh, right — good point.